### DISCUSSION TITLE
Recover Lost Program

### DISCUSSION THREAD

**Discussion by blushingbacon** [pybricks/support#1698] [Q&A]
I was writing a program using code blocks in Pybricks when suddenly it refreshed and all of my programs were gone. I had made a fair amount of progress since I last backed up my files but a pretty current version is sitting on my spike prime hub. Is there a way to recover the lost file from the pybrick website, my computer, or the hub itself?

If not, is there a way to set up some kind of autosave/backup to help mitigate the damage when this random thing happens again?

**Comment by laurensvalk** [comment]
Thanks for posting this. I've never seen this happen.  Even if the app crashes (which is rare), the files should all still be there when you open the app again, but it sounds like this didn't happen in your case. Did you see any errors when this happened? Did it refresh out of the blue?

The program gets stored on the hub in a "compiled" format that the robot understands, but which does not contain the original program I'm afraid.

**Reply by blushingbacon** [reply]
No errors and it just refreshed out of the blue. I was using it on a Windows 11 laptop running as a Chrome progressive web app. I appreciate the response.

**Reply by BertLindeman** [reply]
I had it once in a similar situation when I let chrome on win11 do an update.
Never had problems with that, but then all pybricks (beta and code) programs were gone. 
But also settings like start webpages. Lesson learned do a daily backup to zip, but also before any update of chrome. 😢 

[EDIT] some re-wording

**Reply by Shushmuffleschmuff** [reply]
Currently, I also just had Pybricks delete all of my program files for a competition when i opened it.
